services/repository.py

```diff
@@ -63,7 +64,35 @@ def get_repo_provider_service(
         ignore_installation=False,
         installation_name=installation_name_to_use,
     )
-    token, token_owner = get_repo_appropriate_bot_token(repository, installation_info)
+    print("INSTALLATION INFO", installation_info)
+    try:
+        token, token_owner = get_repo_appropriate_bot_token(
+            repository, installation_info
+        )
+    except RepositoryWithoutValidBotError:
+        installation_info_without_considering_repo = get_owner_installation_id(
+            repository.owner,
+            False,
+            repository=None,
+            ignore_installation=False,
+            installation_name=installation_name_to_use,
+        )
+        print("INSTALLATION INFO (no repo)", installation_info_without_considering_repo)
```

Comment:
the content in these print statements can still remain, but they should probably be `log.info`

Reply:
Nah, I forgot to remove those :E